### ISSUE TITLE
Reference of type 'Microsoft.CodeAnalysis.UnresolvedMetadataReference' is not valid for this compilation.

### ISSUE THREAD

**Issue by fforjan** [dotnet/roslyn-analyzers#2273]
#### Analyzer package
Microsoft.CodeAnalysis.Metrics

#### Package Version
v2.9.1

#### Repro steps
1. Run the program in command line :
"C:\Users\ISYS36005\TFSWorkspaces\Dev\Source\packages\SchneiderElectric.Software.CodeMetrics.2.0.0\Metrics\metrics.exe" /project:"C:\Users\ISYS36005\TFSWorkspaces\Dev\Source\Ourproject\OurProject.csproj" /OUT:"C:\Users\ISYS36005\TFSWorkspaces\Dev\Source\Ourproject\bin\Debug\OurProject.dll.CodeMetrics.xml"

#### Expected behavior
it should generate the code metrics and if failing, give more details about what is wrong !

#### Actual behavior
Loading OurProject.csproj...
Computing code metrics for OurProject.csproj...
Reference of type 'Microsoft.CodeAnalysis.UnresolvedMetadataReference' is not valid for this compilation.
  Parameter name: references[0]

**Comment by fforjan** [comment]
I have isolated the issue:
one of the project references is a C++ CLR project. 
You can reproduce it by building the solution
[IssueGithub.zip](https://github.com/dotnet/roslyn-analyzers/files/3036754/IssueGithub.zip)

Also note that on the command line, msbuild /t:Metrics is failing, I have added it as a post build step.

We are using VS 2017

**Comment by fforjan** [comment]
@sharwell  sorry to ping you but this is a blocker for us to move away from the old code metrics.
Do you know if there is any workaround ?
if not, the next time for us to do an upgrade will be in 3 or 6 months - we are only doing 'tools' upgrade at a beginning of a quarter

**Comment by mavasani** [comment]
I believe this is again an issue with MSBuild workspace unable to load the project/solution, nothing specific to Code Metrics application here. @jinujoseph can we please port this issue to Roslyn repo? I will investigate why we are getting an unresolved metadata reference here..

**Comment by mavasani** [comment]
Ah, seems like you have already found the underlying reason: `one of the project references is a C++ CLR project.`. Note that Code Metrics console app is just using `MSBuildWorkspace.Create` and `MSBuildWorkspace.OpenProjectAsync` APIs to load the project [here](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Tools/Program.cs#L307).

@jasonmalinowski @DustinCampbell - doesn't MSBuildWorkspace support loading projects with C++ CLI project reference?

**Comment by jasonmalinowski** [comment]
There's no fundamental reason it _couldn't_ work, but honestly I'm not sure we have tests so I could imagine it's been busted and nobody noticed until now.

**Comment by fforjan** [comment]
@jasonmalinowski  there is an attachment with a simple project to reproduce it on my first comment

**Comment by fforjan** [comment]
Any update or workaround ?

**Comment by fforjan** [comment]
@jasonmalinowski @mavasani  2.9.4-beta1.final do not fix this ?

**Comment by jasonmalinowski** [comment]
@mavasani Does this need to be moved over to Roslyn?
@fforjan I don't believe this bug has been investigated.

**Comment by mavasani** [comment]
> @mavasani Does this need to be moved over to Roslyn?

Yes, @jinujoseph can you port it to Roslyn repo? I am not an admin on it to do the github port.
